lib/PuppeteerSharp/NetworkManager.cs

```diff
@@ -90,18 +96,33 @@ internal Task SetExtraHTTPHeadersAsync(Dictionary<string, string> extraHTTPHeade
 
         internal async Task SetOfflineModeAsync(bool value)
         {
-            if (_offine != value)
-            {
-                _offine = value;
+            _emulatedNetworkConditions.Offline = value;
+            await _updateNetworkConditions().ConfigureAwait(false);
+        }
 
-                await _client.SendAsync("Network.emulateNetworkConditions", new NetworkEmulateNetworkConditionsRequest
-                {
-                    Offline = value,
-                    Latency = 0,
-                    DownloadThroughput = -1,
-                    UploadThroughput = -1
-                }).ConfigureAwait(false);
-            }
+        internal async Task EmulateNetworkConditions(NetworkConditions networkConditions)
+        {
+            _emulatedNetworkConditions.Upload = networkConditions != null
+              ? networkConditions.Upload
+              : -1;
+            _emulatedNetworkConditions.Download = networkConditions != null
+              ? networkConditions.Download
+              : -1;
+            _emulatedNetworkConditions.Latency = networkConditions != null
+              ? networkConditions.Latency
+              : 0;
+            await _updateNetworkConditions().ConfigureAwait(false);
+        }
+
+        internal async Task _updateNetworkConditions()
+        {
+            await _client.SendAsync("Network.emulateNetworkConditions", new NetworkEmulateNetworkConditionsRequest
+            {
+                Offline = _emulatedNetworkConditions.Offline,
+                Latency = _emulatedNetworkConditions.Latency,
+                UploadThroughput = _emulatedNetworkConditions.Upload,
+                DownloadThroughput = _emulatedNetworkConditions.Download,
+            }).ConfigureAwait(false);
         }
```

Comment:
Let's use an arrow function and return the task instead of making the method async

Reply:
Okay